CodeSignal/dockerfiles
# MySQL upgrade

#57.

__Changes in this and related PRs:__

As mentioned in the issue, I went for the simplest solution: create a new docker container with latest version, and use it for database tasks only. Thus, I made changes to three repos:

1) **dockerfiles:** Created a new container, and removed `main` container since it's not used anymore anyway.
2) **coderunners:** Made some changes to use this new image, and updated `sql_runner.sh` script accordingly. Also added a few tests to verify that things work.
3) **codesignal:** Updated template to stop using deprecated sql_mode.

__How to test this:__

The easiest way to test the changes it to have a coderunner deployed locally on your VM. 

_Setup:_
1) Check out changes made to the coderunner repo to your local VM
2) Build this new docker image on your VM with `docker build -t codesignal/mysql:v1.0 .` while in a folder that contains `Dockerfile`.
3) Check out changes to the main repo
4) Set up your local environment in a way where run requests get routed to your local coderunner
   * Run Rundash locally;
   * Have your local coderunner in Rundash db;
   * Update `rundashService::_callRundash`  method of the main with `{ host: 'localhost' }`as the third `rundashConnect.call` argument.

_Testing:_
1) Verify that the tests work
   * Run `mocha test/raw.mysql.test.js` on your local coderunner
2) Verify that database tasks work
   * Open any DB task that supports mysql, and try submitting solution from content repo.

__Next steps:__

Once the changes are approved, I will:
1) Create new image on dockerhub;
2) Deploy coderunner that users this image;
3) Merge changes to the main app, and add deployment instructions to remind the deployer to deploy new coderunners and enable them right after deploy.
